packages/searchinput/src/react/index.js

```diff
@@ -1,72 +1,84 @@
 import { CloseIcon, SearchIcon } from '@pluralsight/ps-design-system-icon'
 import TextInput from '@pluralsight/ps-design-system-textinput'
+import { callAll } from '@pluralsight/ps-design-system-util'
 
 import stylesheet from '../css/index.js'
 
 const styles = {
-  clear: _ => css(stylesheet['.psds-searchinput-clear']),
+  clear: clearVisible =>
+    css(
+      stylesheet['.psds-searchinput-clear'],
+      clearVisible && stylesheet['.psds-searchinput-clear__visible']
+    ),
   field: _ => css(stylesheet['.psds-searchinput-field'])
 }
 
-const SearchInput = React.forwardRef((props, forwardedRef) => {
-  const { loading, onClear, ...rest } = props
+const SearchInput = React.forwardRef(
+  ({ loading, onClear, onChange, ...rest }, forwardedRef) => {
+    const [clearVisible, toggleClear] = useState(false)
+    const handleChange = e => {
+      e.currentTarget.value.length > 0 ? toggleClear(true) : toggleClear(false)
+    }
+    const ref = useRef()
+    React.useImperativeHandle(forwardedRef, () => ref.current)
 
-  const ref = useRef()
-  React.useImperativeHandle(forwardedRef, () => ref.current)
+    const handleClear = evt => {
+      if (ref.current) {
+        ref.current.value = ''
+        ref.current.focus()
+      }
 
-  const handleClear = evt => {
-    if (ref.current) {
-      ref.current.value = ''
-      ref.current.focus()
+      onClear(evt)
     }
 
-    onClear(evt)
-  }
+    const clearBtn = onClear && (
+      <Button
+        onClick={handleClear}
+        {...styles.clear(clearVisible)}
+        appearance={Button.appearances.flat}
+        icon={<CloseIcon />}
+        iconOnly
+        size={Button.sizes.small}
+      />
+    )
 
-  const clearBtn = onClear && (
-    <Button
-      onClick={handleClear}
-      {...styles.clear()}
-      appearance={Button.appearances.flat}
-      icon={<CloseIcon />}
-      iconOnly
-      size={Button.sizes.small}
-      {...props}
-    />
-  )
+    const icon = loading ? (
+      <CircularProgress size={CircularProgress.sizes.small} />
+    ) : (
+      <SearchIcon />
+    )
 
-  const icon = loading ? (
-    <CircularProgress size={CircularProgress.sizes.small} />
-  ) : (
-    <SearchIcon />
-  )
+    return (
+      <TextInput
+        {...styles.field()}
+        appearance={TextInput.appearances.subtle}
+        fieldAfter={clearBtn}
+        icon={icon}
+        ref={ref}
+        onChange={callAll(handleChange, onChange)}
+        type="search"
+        {...rest}
+      />
+    )
+  }
+)
 
-  return (
-    <TextInput
-      {...styles.field()}
-      appearance={TextInput.appearances.subtle}
-      fieldAfter={clearBtn}
-      icon={icon}
-      ref={ref}
-      type="search"
-      {...rest}
-    />
-  )
-})
+SearchInput.displayName = 'SearchInput'
 
 SearchInput.defaultProps = {
   loading: false
 }
 
 SearchInput.propTypes = {
   loading: PropTypes.bool,
-  onClear: PropTypes.func
+  onClear: PropTypes.func,
+  onChange: PropTypes.func
```

Comment:
What's this for?  Does this address the original legibility problem from Kristie?  If not, why include it?

Reply:
- **onChage**? I just need to destructure that out and decided to change the way the prop destructuring was being done so it required a propType declaration. 
- **onClear** has been part of the component prior to this. 

The reason I desctructured **onChange** out was to address the [interaction](https://github.com/pluralsight/design-system/pull/983#issuecomment-618477680) @sweeting  mentioned above. That functionality was not actually part of the component. These changes address both Brian and Carolina's [accessibility issue](https://pluralsight.slack.com/archives/C70HPSJPP/p1587589348077800) for the original slack post mentioned when I opened the PR.

Reply:
Ah yeah, thank you for helping me remember the interaction in question.  Tried in storybook, looks great.